test/runtests.jl

```diff
@@ -1300,8 +1301,10 @@ end
     @test dot(Zeros(5), Ones{ComplexF16}(5)) ≡ zero(ComplexF64)
     @test dot(Ones{ComplexF16}(5), Zeros(5)) ≡ zero(ComplexF64)
     @test dot(randn(5), Zeros{ComplexF16}(5)) ≡ dot(Zeros{ComplexF16}(5), randn(5)) ≡ zero(ComplexF64)
+    @test dot(c, Fill(1 + im, 15)) ≡ dot(Fill(1 + im, 15), c)' ≈ dot(c, fill(1 + im, 15))
 
     @test dot(Fill(1,5), Fill(2.0,5)) ≡ 10.0
+    @test dot(Fill(true,5), Fill(Int8(1),5)) isa Int8
```

Comment:
Add a matrix case to test:
```julia
julia> dot([[1 2; 3 4]], [[1 2; 3 4]])
30
```
(I think your changes actually fixed a bug here

Reply:
do you mean `@test dot([Fill(1, 2, 2)], [[1 2; 3 4]])`?

Reply:
No I mean something like `dot(Fill([1 2; 3 4],3),  Fill([5 6; 7 8],3))`

Reply:
Can I add `dot` test to `test_addition_and_subtraction`?

Reply:
ok